lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
```

Comment:
Are we sure we don't do this optimization anywhere else in the code more generically? It sounds like something that would go in `util`. If we don't then LGTM.